strings/horspool/horspool.go

```diff
@@ -1,95 +1,60 @@
+// Implementation of the
+// [Boyer–Moore–Horspool algorithm](https://en.wikipedia.org/wiki/Boyer%E2%80%93Moore%E2%80%93Horspool_algorithm)
+
 package horspool
 
-// User defined.
-// Set to true to read input from two command line arguments
-// Set to false to read input from two files "pattern.txt" and "text.txt"
-// const commandLineInput bool = false
+import "errors"
+
+var ErrNotFound = errors.New("pattern was not found in the input string")
+
+func Horspool(t, p string) (int, error) {
+	// in order to handle multy-byte character properly
+	// the input is converted into rune arrays
+	return horspool([]rune(t), []rune(p))
```

Comment:
Implementing the functionality in a separate function seems unnecessary. Just directly implement it here.

Reply:
Introducing `tRunes`/`pRunes` seems fine to me. @siriak what do you think?

Reply:
I prefer the current implementation because it clearly assigns concerns to different functions. The strings function accepts input and converts it to a different format, while the runes function implements the algorithm. Otherwise, there would be `t, p, tRune and pRune` variables that span the whole method, but only `tRune and pRune` variables are used.

Reply:
I'm also kind of against this... but why not just use the `[]rune` on the main `Horspool` instead?

Reply:
For some reason `horspool` is in `strings` directory. I think that `string` is a more _natural_ type to represent string than `[]rune` - I would be annoyed if I would need to convert my inputs to `[]rune` every single time when I want to use this function.

In this case it is a pity that go does not have overloading.

Reply:
> I would be annoyed if I would need to convert my inputs to []rune every single time

I get annoyed at that too 😂 until there are points where the conversion makes sense (not just the `[]rune` conversion but `[]byte` too). In this case however, it's fine I don't mind it. I just didn't like another allocation happening for that function call but that is a very nitpicky 😂

Reply:
> I just didn't like another allocation happening for that function call but that is a very nitpicky 😂

I guess the only solution for that would be to create a `SimpleString` class not allowing multi-byte characters or `TurboString`, which has just a `[]rune` wrapper.

I any case: I will mark this conversation as _resolved_. Thanks for interesting conversation.